magic-training.lic

```diff
@@ -18,58 +18,42 @@ class MagicTraining
-        { name: 'training_skills', regex: /^((Augmentation|Warding|Utility|Sorcery)((\s*,\s*)(Augmentation|Warding|Utility|Sorcery))*)$/i, optional: true, description: 'Override which magic skills to train, as a comma-separated list. (Optional)' }
+        { name: 'training_skills', regex: /^((Augmentation|Warding|Utility|Sorcery)((\s*,\s*)(Augmentation|Warding|Utility|Sorcery))*)$/i, optional: true, description: 'Override which magic skills to train, as a comma-separated list. (Optional)' },
+        { name: 'script_summary', optional: true, description:'Trains magic skills defined in Yaml setting training_skills' }
       ]
     ]
 
     args = parse_args(arg_definitions)
+    max_time = ((args.max_time || 60).to_i * 60) # convert to seconds
+    training_skills = args.training_skills ? args.training_skills.split(',').map { |skill| skill.strip.downcase.capitalize } : %w[Utility Warding Augmentation Sorcery]
 
-    load_settings
-    DRC.wait_for_script_to_complete('buff', ['mana'])
-
-    @max_time = ((args.max_time || 60).to_i * 60) # convert to seconds
-    @training_skills = args.training_skills.split(',').map { |skill| skill.strip.downcase.capitalize }
-    @training_skills = %w[Utility Warding Augmentation Sorcery] if @training_skills.nil? || @training_skills.empty?
-
-    echo "Max training time: #{@max_time} seconds"
-    echo "Training skills: #{@training_skills}"
+    echo "Max training time: #{max_time} seconds"
+    echo "Training skills: #{training_skills}"
 
     loop do
-      if (Time.now - @@time_in) > @max_time
+      if (Time.now - @@time_in) > max_time
         echo "Max time alloted for Magic Training reached. Now exiting..."
         exit
       end
-      unless train_magics?
+      unless train_magics?(training_skills)
         echo "Done training magics"
         exit
       end
     end
   end
 
-  def load_settings
-    @settings = get_settings
-
-    # Stops training a specfic skill after reaching this mindstate threshold, and quits program when all skills are above
-    @exp_threshold = @settings.magic_exp_training_max_threshold
-
-    # Hash of magic skill (e.g. Warding) to spell to train with (same config as spells in waggle sets)
-    @training_spells = @settings.magic_training # original value from `;repos download magic-training`
-    @training_spells = @settings.training_spells if @training_spells.nil? || @training_spells.empty?
-
-    @force_cambrinth = @settings.waggle_force_cambrinth
-
-    @health_threshold = @settings.health_threshold # vitaltiy percentage
-    @saferoom_health_threshold = @settings.saferoom_health_threshold # wound severity threshold
-  end
-
-  def train_magics?
-    load_settings # This is done purposely so you can adjust settings on the fly, it's great for finding mana values
+  def train_magics?(training_skills)
+    settings = get_settings
+    DRCT.walk_to(settings.magic_training_room)
```

Comment:
What happens if `magic_training_room` isn't specified? Real question, don't know how `DRCT.walk_to("")` responds.

Reply:
Checked that, it just returns, doesnt even open go2

Reply:
Return unless room, iirc

Reply:
> 

I don't use this script, even historically. What was prior behavior. Did it just train in the room you were standing in?

Reply:
Yes.

Reply:
Is this comment resolved?